plugins/browser-plugin-media-tracking/src/index.ts

```diff
@@ -0,0 +1,230 @@
+      eventHandlers[ev] = (el: HTMLAudioElement | HTMLVideoElement) => eventsWithOtherFunctions[ev](el);
+    }
+    eventHandlers[ev] = (el: HTMLAudioElement | HTMLVideoElement, e: MediaEventType) => mediaPlayerEvent(el, e, conf);
+  }
+
+  trackedIds[args.id] = { searchIntervals: [], searchLimit: 5, tracking: false };
+  setUpListeners(args.id, conf, eventHandlers);
+  trackedIds[args.id].searchIntervals.push(setInterval(() => setUpListeners(args.id, conf, eventHandlers), 5000));
+}
+
+function setUpListeners(id: string, conf: TrackingOptions, eventHandlers: Record<string, Function>) {
+  // The element may not be loaded in time for this function to run,
+  // so we have a few goes at finding the element
+  let el = findMediaElem(id);
+
+  if (!trackedIds[id].searchLimit) {
+    console.error("Couldn't find element before timeout");
+    trackedIds[id].searchIntervals.forEach((e: ReturnType<typeof setTimeout>) => clearInterval(e));
+    return;
+  }
+
+  if (!el) {
+    trackedIds[id].searchLimit--;
+    console.error(`Couldn't find element with id: '${id}'. Trying again...`);
+    return;
+  }
+
+  trackedIds[id].searchIntervals.forEach((e: ReturnType<typeof setTimeout>) => clearInterval(e));
+  if (!trackedIds[id].tracking) {
+    if (conf.captureEvents.indexOf(SnowplowMediaEvent.PERCENTPROGRESS) !== 0) {
+      boundryErrorHandling(conf.progress!.boundries);
+      setPercentageBoundTimeouts(el, conf);
+    }
+    addCaptureEventListeners(el, conf.captureEvents, eventHandlers);
+    trackedIds[id].tracking = true;
+  }
+}
+
+function addCaptureEventListeners(
+  el: HTMLAudioElement | HTMLVideoElement,
+  captureEvents: EventGroup,
+  eventHandlers: Record<string, Function>
+): void {
+  for (let e of captureEvents) {
+    let ev: EventListener = () => eventHandlers[e](el, e);
+    if (isTypeTextTrackEvent(e)) {
+      el.textTracks.addEventListener(e, ev);
+    } else if (isTypeDocumentEvent(e)) {
+      document.addEventListener(e, ev);
+      // Chrome and Safari both use the 'webkit' prefix for the 'fullscreenchange' event
+      // IE uses 'MS'
+      if (e === DocumentEvent.FULLSCREENCHANGE) {
+        document.addEventListener('webkit' + e, ev);
+        document.addEventListener('MS' + e, ev);
+      }
+    } else {
+      el.addEventListener(e, ev);
+    }
+  }
+}
+
+function mediaPlayerEvent(
+  el: HTMLAudioElement | HTMLVideoElement,
+  e: MediaEventType,
+  conf: TrackingOptions,
+  boundry?: number
+): void {
+  let event = buildMediaEvent(el, e, conf.label, boundry);
+  if (conf.captureEvents.indexOf(SnowplowMediaEvent.PERCENTPROGRESS) !== -1) {
+    progressHandler(e, el, conf);
+  }
+
+  // Dragging the volume scrubber will generate a lot of events, this limits the rate at which
+  // volume events can be sent at
+  if (e === MediaEvent.VOLUMECHANGE) {
+    clearTimeout(Number(conf.volumeChangeTimeout));
+    conf.volumeChangeTimeout = setTimeout(() => trackMediaEvent(event), 200);
```

Comment:
Time for that classic PR review comment... should `200` be configurable? `volumeChangeTrackingInterval` or something?

Reply:
If not, maybe 200 is too low? Would 500 be better?